File: packages/postpin-go/serviceability.go
package postpin

import (
	"context"
	"net/http"
	"net/url"
)

// ServiceabilityService checks pincode serviceability.
type ServiceabilityService struct{ client *Client }

// Check reports whether a pincode is serviceable.
func (s *ServiceabilityService) Check(ctx context.Context, pincode string) (*Serviceability, error) {
	data, err := s.client.do(ctx, http.MethodGet, "/public/serviceability/"+url.PathEscape(pincode), nil, nil, "")
	if err != nil {
		return nil, err
	}
	var out Serviceability
	if err := decode(data, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

File: packages/postpin-php/src/Resource/Serviceability.php
<?php

declare(strict_types=1);

namespace Postpin\Resource;

use Postpin\Http\HttpClient;

/** Pincode serviceability checks. */
final class Serviceability
{
    public function __construct(private readonly HttpClient $http)
    {
    }

    /**
     * Check whether a pincode is serviceable.
     *
     * @return array<string, mixed>
     */
    public function check(string $pincode): array
    {
        return (array) $this->http->request('GET', '/public/serviceability/' . rawurlencode($pincode));
    }
}
